File: Codes from the AI-assistants/[CLBG] regex-redux/Specialized Prompt - TBR Mode/RegexRedux-gemini-specialized-tbr.py
import sys
import re

def main():
    data = sys.stdin.read()
    ilen = len(data)
    
    seq = re.sub(r'>[^\n]*\n', '', data)
    seq = seq.replace('\n', '')
    clen = len(seq)
    
    patterns = [
        r'agggtaaa|tttaccct',
        r'[cgt]gggtaaa|tttaccc[acg]',
        r'a[act]ggtaaa|tttacc[agt]t',
        r'ag[act]gtaaa|tttac[agt]ct',
        r'agg[act]taaa|ttta[agt]cct',
        r'aggg[acg]aaa|ttt[cgt]ccct',
        r'agggt[cgt]aa|tt[acg]accct',
        r'agggta[cgt]a|t[acg]taccct',
        r'agggtaa[cgt]|[acg]ttaccct'
    ]
    
    counts = []
    for pat in patterns:
        cnt = len(re.findall(pat, seq))
        counts.append((pat, cnt))
    
    magic_patterns = [
        (r'tHa[Nt]', r'<4>'),
        (r'aND|caN|Ha[DS]|WaS', r'<3>'),
        (r'a[NSt]|BY', r'<2>'),
        (r'<[^>]*>', r'|'),
        (r'\|[^|][^|]*\|', r'-')
    ]
    
    current = seq
    for pat, repl in magic_patterns:
        current = re.sub(pat, repl, current)
    flen = len(current)
    
    for pat, cnt in counts:
        print(f"{pat} {cnt}")
    
    print()
    print(ilen)
    print(clen)
    print(flen)

if __name__ == '__main__':
    main()

File: Codes from the AI-assistants/[Experimental] regex-redux/Specialized Prompt - TBR Mode/RegexRedux-gemini-specialized-tbr.go
variants := []string{
    "agggtaaa|tttaccct",
    "[cgt]gggtaaa|tttaccc[acg]",
    "a[act]ggtaaa|tttacc[agt]t",
    "ag[act]gtaaa|tttac[agt]ct",
    "agg[act]taaa|ttta[agt]cct",
    "aggg[acg]aaa|ttt[cgt]ccct",
    "agggt[cgt]aa|tt[acg]accct",
    "agggta[cgt]a|t[acg]taccct",
    "agggtaa[cgt]|[acg]ttaccct",
}

replacements := []struct{ p, r string }{
    {"tHa[Nt]", "<4>"},
    {"aND|caN|Ha[DS]|WaS", "<3>"},
    {"a[NSt]|BY", "<2>"},
    {"<[^>]*>", "|"},
    {"\\|[^|][^|]*\\|", "-"},
}

// Pseudocode
for _, repl := range replacements {
    re := regexp.MustCompile(repl.p)
    sequenceForReplacement = re.ReplaceAll(sequenceForReplacement, []byte(repl.r))
}